### PR TITLE
Add missing metadata to eamxx output files

### PR DESCRIPTION
**Pull request by bartgol**
Metadata that was missing included IC/topo file names, user/host names, git/model version, creation date, and case id.

This should complete the [list of IO tasks](https://acme-climate.atlassian.net/wiki/spaces/NGDNA/pages/3290497244/SCREAMv1+Output+Netcdf+Cleanup) we had on confluence. We still have the PR for scorpio clib open, but that's not a priority, so it wasn't on that list.

### CODE REVIEW

**Comment by AaronDonahue** [comment]
@golaz and @czender do either of you have any comments regarding the metadata itself?  See for example Luca's question above.

File: components/eamxx/src/share/io/scream_output_manager.cpp [hunk 1]
```diff
@@ -694,26 +693,36 @@ setup_file (      IOFileSpecs& filespecs,
...
+  set_attribute<std::string>(filename,"source","E3SM Atmosphere Model (EAMxx)");
+  set_attribute<std::string>(filename,"eamxx_version",EAMXX_VERSION);
+  set_attribute<std::string>(filename,"git_version",p.get<std::string>("git_version",EAMXX_GIT_VERSION));
+  set_attribute<std::string>(filename,"hostname",p.get<std::string>("hostname","UNKNOWN"));
+  set_attribute<std::string>(filename,"username",p.get<std::string>("username","UNKNOWN"));
+  set_attribute<std::string>(filename,"atm_initial_conditions_file",p.get<std::string>("initial_conditions_file","NONE"));
+  set_attribute<std::string>(filename,"topography_file",p.get<std::string>("topography_file","NONE"));
+  set_attribute<std::string>(filename,"contact","e3sm-data-support@listserv.llnl.gov");
+  set_attribute<std::string>(filename,"institution_id","E3SM-Projet");
+  set_attribute<std::string>(filename,"product",(m_is_model_restart_output ? "model-restart" : "model-output"));  // TODO
```
Comment (AaronDonahue) [inline components/eamxx/src/share/io/scream_output_manager.cpp:723]: Is this still "TODO", I see the "TODO" comment at the end of line is still there.

**Comment by czender** [comment]
@bartgol, @AaronDonahue, and @golaz this PR certainly looks like a big step in the right direction. The global attributes shown in Luca's question above seem to be mainly CMIP and/or ESGF-related and are outside my realm of experience (which is mainly CF Conventions and per-variable physics attributes). Those attributes above that I do understand (e.g., `Conventions`) look good to me. I suggest asking @chengzhuzhang to vet the rest. And I do not know the answers to Luca's explicit questions above.

**Comment by chengzhuzhang** [comment]
> 
> Things that confused me on the confluence page:
> 
>     * what's the difference between `source_id` and `git_hash`?
> 
>     * what's the difference between `product` and `title`? Should product be a "product type" (restart vs output), and title reflect the prefix of the output stream maybe?

Thanks for looping me in. 
I looked up an example EAM output for attribute for here mentioned:
```
// global attributes:
		:ne = 0 ;
		:fv_nphys = 2 ;
		:title = "EAM History file information" ;
		:source = "E3SM Atmosphere Model" ;
		:source_id = "b936852235" ;
		:product = "model-output" ;
		:realm = "atmos" ;
		:case = "v2.NARRM.historical_0101" ;
		:username = "ac.qtang" ;
		:hostname = "chrysalis" ;
		:git_version = "b936852235" ;
		:history = "created on 11/04/21 03:21:43" ;
		:Conventions = "CF-1.7" ;
		:institution_id = "E3SM-Project" ;
		:institution = "LLNL (Lawrence Livermore National Laboratory, Livermore, CA 94550, USA); ANL (Argonne National Laboratory, Argonne, IL 60439, USA); BNL (Brookhaven National Laboratory, Upton, NY 11973, USA); LANL (Los Alamos National Laboratory, Los Alamos, NM 87545, USA); LBNL (Lawrence Berkeley National Laboratory, Berkeley, CA 94720, USA); ORNL (Oak Ridge National Laboratory, Oak Ridge, TN 37831, USA); PNNL (Pacific Northwest National Laboratory, Richland, WA 99352, USA); SNL (Sandia National Laboratories, Albuquerque, NM 87185, USA). Mailing address: LLNL Climate Program, c/o David C. Bader, Principal Investigator, L-103, 7000 East Avenue, Livermore, CA 94550, USA" ;
		:contact = "e3sm-data-support@listserv.llnl.gov" ;
		:initial_file = "v2.NARRM.piControl.eam.i.0101-01-01-00000.nc" ;
		:topography_file = "/lcrc/group/e3sm/data/inputdata/atm/cam/topo/USGS_northamericax4v1pg2_12xdel2_consistentSGH_20020209.nc" ;
		:time_period_freq = "month_1" ;
}
```
Among the global attributes above, `source_id` and `git_version` are duplicated.   The `source_id` is also a CMIP attribute, CMIP require to add additional global attribute based on information E3SM provided when register the simulations. Example :source_id = "E3SM-2-0". The value can be assigned after a model version released. I don’t think we need `source_id` in native output.

For other 3 attribute, `git_hash`, `product` and `title` are inherited from EAM. I'm not very familiar with the convention, maybe @rljacob can chime in here.

File: components/eamxx/src/share/io/scream_output_manager.cpp [hunk 2]
```diff
@@ -694,26 +693,36 @@ setup_file (      IOFileSpecs& filespecs,
...
-  set_attribute<std::string>(filename,"product","");  // TODO
-  set_attribute<std::string>(filename,"component","ATM");
+  set_attribute<std::string>(filename,"source","E3SM Atmosphere Model (EAMxx)");
+  set_attribute<std::string>(filename,"eamxx_version",EAMXX_VERSION);
+  set_attribute<std::string>(filename,"git_version",p.get<std::string>("git_version",EAMXX_GIT_VERSION));
+  set_attribute<std::string>(filename,"hostname",p.get<std::string>("hostname","UNKNOWN"));
+  set_attribute<std::string>(filename,"username",p.get<std::string>("username","UNKNOWN"));
+  set_attribute<std::string>(filename,"atm_initial_conditions_file",p.get<std::string>("initial_conditions_file","NONE"));
+  set_attribute<std::string>(filename,"topography_file",p.get<std::string>("topography_file","NONE"));
+  set_attribute<std::string>(filename,"contact","e3sm-data-support@listserv.llnl.gov");
```
Comment (chengzhuzhang) [inline components/eamxx/src/share/io/scream_output_manager.cpp:721]: Please change `contact` to: `e3sm-data-support@llnl.gov` . The one with `listserv` will not be supported some time in the future.

**Comment by rljacob** [comment]
Yes "source_id" is there for CMIP to later overwrite it.  If we don't need it, git rid of it.

I think "product" is also a CMIP thing that we are filling out ahead of time so there's less work later.

"title" was inherited from CAM.

**Review by chengzhuzhang — APPROVED** [review]
The updates on metadata for global attribute look good to me. Thank you!

**Review by AaronDonahue — APPROVED** [review]

**Pull request merged**